### PR TITLE
*: support listener

### PR DESCRIPTION
**Pull request by BusyJay**
This PR adds listener wrappers. The main difference is using direct pointers to various infos instead of a wrapper type.

### CODE REVIEW

File: tirocks/src/listener.rs [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+}
+
+extern "C" fn destructor<E: EventListener>(ctx: *mut c_void) {
+    unsafe {
+        drop(Box::from_raw(ctx as *mut E));
+    }
+}
+
+// Maybe we should reuse db instance?
+// TODO: refactor DB implement so that we can convert DBInstance to DB.
```
Comment (tabokie) [inline tirocks/src/listener.rs:121]: I don't understand this comment. `on_flush_begin` will accept a DB after we rust-ify the DB type?

Comment (BusyJay) [reply tirocks/src/listener.rs:121]: Yes. More precisely, there will be no `crocksdb_t`, and it will accept `*mut rocksdb_DB` directly.

**Review by tabokie — APPROVED** [review]

**Comment by BusyJay** [comment]
@Connor1996 PTAL.

A full version of the latest (all) changes can be found at branch dev. https://github.com/BusyJay/tirocks/tree/dev

**Review by Connor1996 — APPROVED** [review]
LGTM

**Pull request merged**